### DISCUSSION TITLE
What is the difference between a web implementation of Firebase analytics?

### DISCUSSION THREAD

**Discussion by braincomb** [capawesome-team/capacitor-firebase#92] [General]
Hi Robin,

I am looking to integrate Firebase analytics into my Capacitor app and saw that there are some Capacitor plugins such as this one.

My question is, since a Capacitor app is basically a WebView on the client facing side, could we not just simply use the web implementation of Firebase analytics, i.e.:
```js
import { getAnalytics } from "firebase/analytics";
const firebaseApp = initializeApp(firebaseConfig);
const analytics = getAnalytics(firebaseApp);
```
And then just call web methods.

Is there any added benefit to using a Capacitor plugin specifically?

Appreciate any insight.

Thanks!

**Comment by robingenz** [comment]
Hi @braincomb,
It depends on what you want to do with Firebase Analytics.
Basically, the Firebase JavaScript SDK is all you need.
However, as far as I know, additional data from other Firebase products is also collected via Firebase Analytics.
For example, from Firebase Cloud Messaging (see [docs](https://firebase.google.com/docs/cloud-messaging/understand-delivery?platform=android)).
Also, I'm not sure if the native SDKs on Android and iOS handle states like no network connection etc. better.

I hope this helps you a bit.
